### PR TITLE
[Mobile SuperRes] Update ort session creation to app onCreate() stage

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
- Changes to avoid session creation happening with every instance instantiation.

- Other minor changes

### CODE REVIEW

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt [hunk 1]
```diff
@@ -30,8 +30,11 @@ class MainActivity : AppCompatActivity() {
         outputImage = findViewById(R.id.imageView2);
         superResolutionButton = findViewById(R.id.super_resolution_button)
 
+        val sessionOptions: OrtSession.SessionOptions = OrtSession.SessionOptions()
+        sessionOptions.registerCustomOpLibrary(OrtxPackage.getLibraryPath())
+
         superResolutionButton?.setOnClickListener {
-            performSuperResolution()
+            performSuperResolution(ortEnv.createSession(readModel(), sessionOptions))
```
Comment (natke) [inline mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt:37]: Trying to understand the scoping here: so the createSession() is called when the setOnClickListener is declared?

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt [hunk 2]
```diff
@@ -27,36 +26,32 @@ internal class SuperResPerformer(
...
+            output.use {
+                val rawOutput = (output?.get(0)?.value) as ByteArray
+                val outputImageBitmap =
+                    byteArrayToBitmap(rawOutput)
+
+                // Step 5: set output result
+                result.outputBitmap = outputImageBitmap
             }
         }
+        ortSession.close()
```
Comment (natke) [inline mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt:49]: Wouldn't we want the session to stay open for subsequent calls?

**Review by natke — COMMENTED** [review]
Do we also need to make these changes for the iOS app?

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt (hunk 2 above)
Comment (skottmckay) [reply mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt:49]: Yeah - the examples should make it clear you want to create the session once and keep it around as long as you want to do any inferencing. It's an extremely expensive operation, and OrtSession.run is designed to be called multiple times including concurrently.

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt [hunk 3]
```diff
@@ -30,8 +30,11 @@ class MainActivity : AppCompatActivity() {
         outputImage = findViewById(R.id.imageView2);
         superResolutionButton = findViewById(R.id.super_resolution_button)
 
+        val sessionOptions: OrtSession.SessionOptions = OrtSession.SessionOptions()
+        sessionOptions.registerCustomOpLibrary(OrtxPackage.getLibraryPath())
+
```
Comment (skottmckay) [inline mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt:35]: As this is an example we should have comments explaining why we're doing things as the reader is unlikely to be familiar with ORT and this is how we can teach them what to do.

e.g. 'Register the onnxruntime extensions package that contains the custom operators. These are used to decode the input image into the format the original model requires, and to encode the model output into png format'

Can we also use the new 'registerCustomOpsUsingFunction' function so the examples are consistent across platforms?

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt (hunk 2 above)
Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt:49]: > Do we also need to make these changes for the iOS app?

For the ios app, it's probably fine.  The inference session is only being created once.

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt (hunk 1 above)
Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt:37]: sorry - didn't mean to put the create session here. will move it out of this, keep it in OnCreate() along with the registerCustomop steps.

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt (hunk 3 above)
Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/MainActivity.kt:35]: Make sense. thanks for the suggestion. will add comments and notes here.

> Can we also use the new 'registerCustomOpsUsingFunction' function so the examples are consistent across platforms?

Regarding this, yes we should definitely update to this method for registering the custom ops. 

haven't get it tested yet (for iOS as well.) will update.

**Review by natke — APPROVED** [review]

File: mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt (hunk 2 above)
Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt]: moved session.close() to `OnDestory(). ` thanks for the catch.

**Pull request merged**